### DISCUSSION TITLE
user is undefined

### DISCUSSION THREAD

**Discussion by mehradamiri** [pocketbase/pocketbase#3144] [Q&A] [answered]
I'm using pocketbase sveltekit project im trying to get user data ssr I follow a guide but  i think I did something wrong because when I'm trying to get user it will work like this 

```js
export let data;
console.log(data.user); // { user: undefined } 
```
# hooks.server.js
```js
import PocketBase from 'pocketbase';
import { serializeNonPOJOs } from '$lib/serialize';

export const handle = async ({ event, resolve }) => {
	event.locals.pb = new PocketBase('https://lotus.iran.liara.run');
	event.locals.pb.authStore.loadFromCookie(event.request.headers.get('cookie') || '');

	if (event.locals.pb.authStore.isValid) {
		event.locals.user = serializeNonPOJOs(event.locals.pb.authStore.model);
	} else {
		event.locals.user = undefined;
	}

	const response = await resolve(event);

	response.headers.set('set-cookie', event.locals.pb.authStore.exportToCookie({ secure: false }));

	return response;
};

```
# +layout.server.js

```js
export const load = ({ locals }) => {
	if (locals.user) {
		return {
			user: locals.user
		};
	}

	return {
		user: undefined
	};
};

```

**Comment by ganigeorgiev** [comment, marked as answer]
Are you sure that `event.locals.user` (and `event.locals.pb.authStore.model`) is actually set?
If not, try to add a debug marker or console.log to ensure that the cookie was properly loaded.

Additionally, keep in mind that if you are planning to perform some Node.js access controls based on the `event.locals.user` value **you can't trust it** because the cookie can be spoofed and you need to verify its loaded token and user data against the PocketBase server (in the above non of the code samples sends requests to PocketBase).
To do that you can call `event.locals.pb.collection("users").authRefresh()`. On success this call will automatically populate the auth store with the latest user data and a new token with refreshed `exp` claim. Otherwise, on invalid or expired token - throws an error.
_You can also check the SvelteKit example in https://github.com/pocketbase/js-sdk#ssr-integration._
